internal/user/handler.go

```diff
@@ -21,13 +22,40 @@ func GetFromContext(ctx context.Context) (*User, bool) {
 	return userData, ok
 }
 
+func ConvertEmailsToSlice(emails interface{}) []string {
```

Comment:
There’s a redundant nil check here. When the input is nil, slice, ok := emails.([]string) will already return ok = false.
You can simplify the logic using a switch like this:
```go
func ConvertEmailsToSlice(emails interface{}) []string {
	switch v := emails.(type) {
	case []string:
		if v == nil {
			return []string{}
		}
		return v
	default:
		return []string{}
	}
}
```